splice_si_api/src/main/java/com/splicemachine/si/impl/server/SITransactor.java

```diff
@@ -574,72 +588,156 @@ private ConflictResults checkCommitTimestampForConflict(TxnView updateTransactio
+        if(shouldIgnoreConflictsInternal(table)) {
+            if(LOG.isTraceEnabled()) {
+                LOG.trace(String.format("detection of conflicting data active txns ignored for txn %d as it affects table %s",
+                                        txn.getTxnId(),
+                                        table.getTableName()));
+            }
+            return null;
+        }
+        DataGet dataGet = opFactory.newGet(kvPair.getRowKey(), null);
+        dataGet.addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.TOMBSTONE_COLUMN_BYTES);
+        dataGet.addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.PACKED_COLUMN_BYTES);
+        if(kvPair.getType() == KVPair.Type.DELETE) {
+            dataGet.addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.FK_COUNTER_COLUMN_BYTES);
+        }
+        dataGet.returnAllVersions();
+        DataResult result = table.get(dataGet, null);
+        Iterator<DataCell> iterator = result.iterator();
+        Set<Long> activeTx = new HashSet<>();
+        while(iterator.hasNext()) {
+            long dataTxnId = iterator.next().version();
+            if((dataTxnId & SIConstants.TRANSANCTION_ID_MASK) == txn.getTxnId()) {
+                continue;
+            }
+            // get the parent
+            TxnView txnView = txnSupplier.getTransaction(dataTxnId);
+            long id = txnView.getTxnId();
+            while(true) {
+                TxnView parent = txnView.getParentTxnView();
+                if(parent == Txn.ROOT_TRANSACTION) break;
+                txnView = parent;
+            }
+            Txn.State state = txnView.getState();
+            if(state == Txn.State.ACTIVE) {
+                activeTx.add(txnView.getTxnId() & SIConstants.TRANSANCTION_ID_MASK);
+                if(LOG.isTraceEnabled()) {
+                    String tableName = table.getTableName();
+                    if(table.getDescriptor() != null && table.getDescriptor().getValue("tableDisplayName") != null) {
+                        tableName = table.getDescriptor().getValue("tableDisplayName");
+                    }
+                    LOG.trace(String.format("transaction %d (%d) conflicts with %d on table %s on cell: key [%s] value [%s]",
+                                            txnView.getTxnId() & SIConstants.TRANSANCTION_ID_MASK, id, txn.getTxnId(),
+                                            tableName, Arrays.toString(cell.key()), Arrays.toString(cell.value())));
+                }
+            } else if(state == Txn.State.COMMITTED && txnView.getCommitTimestamp() > txn.getBeginTimestamp()) { // bail out
+                throwWriteWriteConflict(txn, cell, dataTxnId);
+            }
+        }
+
+        return Longs.toArray(activeTx);
+    }
+
+    private static boolean shouldIgnoreConflictsInternal(Partition table) throws IOException {
+        String[] ignoredTables = new String[]{"SYSSTATEMENTS", "SYSCOLUMNS", "SYSDEPENDS"};
```

Comment:
What's the reason for ignoring conflicts on these tables?  Do we do this elsewhere in the code?  Maybe that's why we keep getting corruption on these tables.  Maybe add a comment about why this is needed.

Reply:
Currently we ignore WW conflicts if they happen in certain scenarios:
- calling stored procedures causes writes in `SYSCOLUMNS` and `SYSDEPENDS` (for more information please check Jira DB-3386).
- calling an SPS could cause a write in `SYSSTATEMENTS`
these writes could conflicts, and the conflicts are ignored in these cases. With immediate conflict detection it is easily possible to localize this behavior by surrounding the code path with try-catch-ErrorState-WWConflict-and-ignore. However, with deferred conflict detection it is not possible to follow the same logic since we gather conflicts and check them against at commit-time, so I added this method to do exactly that. I am not claiming this is the right approach, but that's the easiest approach I could think of, I am open to suggestions on how to improve it.

I documented the method accordingly.